src/scenic/core/regions.py

```diff
@@ -2200,6 +2224,199 @@ def dilation(self, iterations, structure=None):
         return VoxelRegion(voxelGrid=new_voxel_grid)
 
+    @cached_property
+    def mesh(self):
+        """(WIP) Return a MeshVolumeRegion representation of this region.
+
+        NOTE: This region is a WIP and will sometimes return None if the transformation
+        is not feasible.
+        """
+        # Extract values for original voxel grid and the surface of the voxel grid.
+        dense_encoding = self.voxelGrid.encoding.dense
+        hpitch = self.voxelGrid.pitch[0] / 2
+        hollow_vr = trimesh.voxel.VoxelGrid(
+            trimesh.voxel.morphology.surface(self.voxelGrid.encoding),
+            transform=self.voxelGrid.transform,
+        )
+
+        surface_indices = numpy.argwhere(hollow_vr.encoding.dense == True)
+        surface_centers = hollow_vr.indices_to_points(hollow_vr.sparse_indices)
+
+        # Determine which faces should be added for each voxel in our extracted surface.
+        point_face_mask_list = []
+
+        def index_in_bounds(index):
+            return all((0, 0, 0) <= index) and all(index < dense_encoding.shape)
+
+        def actual_face(index):
+            return (
+                not index_in_bounds(target_index)
+                or not dense_encoding[tuple(target_index)]
+            )
+
+        for i in range(len(surface_indices)):
+            base_index = surface_indices[i]
+            base_center = surface_centers[i]
+
+            face_mask = [0] * 6
+
+            # Right
+            target_index = numpy.asarray(
+                [base_index[0] + 1, base_index[1], base_index[2]]
+            )
+            if actual_face(target_index):
+                face_mask[0] = True
+
+            # Left
+            target_index = numpy.asarray(
+                [base_index[0] - 1, base_index[1], base_index[2]]
+            )
+            if actual_face(target_index):
+                face_mask[1] = True
+
+            # Front
+            target_index = numpy.asarray(
+                [base_index[0], base_index[1] + 1, base_index[2]]
+            )
+            if actual_face(target_index):
+                face_mask[2] = True
+
+            # Back
+            target_index = numpy.asarray(
+                [base_index[0], base_index[1] - 1, base_index[2]]
+            )
+            if actual_face(target_index):
+                face_mask[3] = True
+
+            # Top
+            target_index = numpy.asarray(
+                [base_index[0], base_index[1], base_index[2] + 1]
+            )
+            if actual_face(target_index):
+                face_mask[4] = True
+
+            # Bottom
+            target_index = numpy.asarray(
+                [base_index[0], base_index[1], base_index[2] - 1]
+            )
+            if actual_face(target_index):
+                face_mask[5] = True
+
+            point_face_mask_list.append((base_center, face_mask))
+
+        # Construct triangles for mesh
+        triangles = []
+
+        for base_center, face_mask in point_face_mask_list:
```

Comment:
Since `point_face_mask_list` isn't used after this, why not just combine these 2 loops, eliminate `face_mask`, and directly append to `triangles`?

Reply:
Good point. I thought we would need to do further filtering but it turned out to be unnecessary.